perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/GroupsManagerMethod.java

```diff
@@ -314,21 +316,108 @@ public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
 		}
 	},
 
+	/*#
+	 * Removes members from a group.
+	 * Non-empty list of members expected. In case of empty list, no member is removed from the group.
+	 * If member is not in the group or the membership is indirect, it is skipped but the rest of the members are processed.
+	 *
+	 * @throw MemberNotExistsException When member doesn't exist
+	 * @throw NotGroupMemberException  When member is not in the group
+	 * @throw GroupNotExistsException When the group doesn't exist
+	 * @throw WrongAttributeAssignmentException When assigning atribute to wrong entity
+	 * @throw AttributeNotExistsException When attribute doesn't exist
+	 * @throw ExternallyManagedException When the group is externally managed
+	 *
+	 * @param group int Group <code>id</code>
+	 * @param members List<Integer> Members - can be empty <code>id</code>
+	 */
+	removeMembers {
+		@Override
+		public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
+			ac.stateChangingCheck();
+			List<Integer> memberInts = parms.readList("members", Integer.class);
+			if (memberInts == null) {
+				throw new RpcException(RpcException.Type.MISSING_VALUE, "Non-empty list of members not sent.");
+			}
+			Group group = ac.getGroupById(parms.readInt("group"));
+			List<Member> members = new ArrayList<>();
+			for (Integer memberInt : memberInts) {
+				try {
+					Member member = ac.getGroupsManager().getGroupMemberById(ac.getSession(), group, memberInt);
+					if (MembershipType.DIRECT.equals(member.getMembershipType())) {
+						members.add(member);
+					}
+				} catch(NotGroupMemberException e) {
+					//just skipped
```

Comment:
Please, write also the reason for skipping there "skipped because user is not member of this group so we don't need to remove him".